rotkehlchen/inquirer.py

```diff
@@ -257,20 +331,38 @@ def find_usd_price(asset: Asset, ignore_cache: bool = False) -> Price:
             return price
 
         price = Price(ZERO)
-        if Inquirer()._cryptocompare.rate_limited_in_last() is False:
-            try:
-                price = Inquirer()._cryptocompare_find_usd_price(asset)
-            except RemoteError:
-                pass
+        for oracle, oracle_instance in Inquirer()._oracle_instances:
+            if oracle_instance.rate_limited_in_last() is True:
+                log.debug(
+                    f"Current price oracle {oracle} can't be used for querying "
+                    f'the USD price for {asset.identifier} due to rate limits. Skipping',
+                )
+                continue
 
-        if price == Price(ZERO):
             try:
-                price = Inquirer()._coingecko.simple_price(from_asset=asset, to_asset=A_USD)
-            except RemoteError as e:
+                price = oracle_instance.query_current_price(
```

Comment:
Standardised method for querying the current USD price via current price oracles.
It is backed by a test that checks all current price oracles implement this method.